### PR TITLE
Add -packageAuthorizationProvider flag for xcodebuild

### PR DESCRIPTION
**Pull request by yarneo**
### Checklist

- [X] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [X] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [X] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [X] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I've updated the documentation if necessary.
- [X] I've added or updated relevant unit tests.

### Motivation and Context

With the new Xcode 16 RC, we have seen hangs in our project when resolving dependencies due to a new flag to xcodebuild that chooses the correct packageAuthorizationProvider when checking out packages.

Quoting from Apple's Xcode 16 RC release notes:
"Fixed: The new xcodebuild flag of -packageAuthorizationProvider keychain|netrc can now be used to force usage of keychain or netrc as the authorizaion provider. This improves Xcode’s package registry support. (118898849)"
[Xcode 16 RC Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-16-release-notes#Resolved-Issues)

This adds an option for users to choose what authorization provider (i.e. netrc / keychain). This resolves all our hang issues and has been tested locally in our product.

### Description

This PR adds a packageAuthorizationProvider option for xcodebuild to set a string with the correct authorization provider. This can be used like so:

```
run_tests(
        project: "Example.xcodeproj",
       ...
        package_authorization_provider: "netrc"   
    )
```

### Testing Steps

Please run an example app and use `package_authorization_provider: "netrc"` as an option, and make sure that when resolving dependencies, there is the `-packageAuthorizationProvider netrc` as part of the xcodebuild:
`xcodebuild -resolvePackageDependencies -scheme Foo -project Example.xcodeproj -packageAuthorizationProvider netrc`

### CODE REVIEW

**Comment by yarneo** [comment]
@joshdholtz @rogerluan would love your review and a potential patch release when you have the time; without this fix, it is currently hanging our CI, and i'm sure others will be coming out the woodwork soon when they move to Xcode 16 if they have private frameworks being fetched with SPM that need authorization.

File: snapshot/lib/snapshot/options.rb [hunk 1]
```diff
@@ -267,6 +267,11 @@ def self.plain_options
                                      description: "Prevents packages from automatically being resolved to versions other than those recorded in the `Package.resolved` file",
                                      type: Boolean,
                                      default_value: false),
+        FastlaneCore::ConfigItem.new(key: :package_authorization_provider,
+                                     env_name: "PACKAGE_AUTHORIZATION_PROVIDER",
```
Comment (revolter) [inline snapshot/lib/snapshot/options.rb:271]:
```suggestion
                                     env_name: "SNAPSHOT_PACKAGE_AUTHORIZATION_PROVIDER",
```

**Review by revolter — CHANGES_REQUESTED** [review]
Thank you for fixing this issue 🙏🏻

Due to time constraints, I didn't actually download Xcode 16 and test this, but having a test and just by looking at the changes, I'm pretty confident that there should be no breaking changes, as it only adds a new and optional option (😅) to `gym`, `scan`, and `snapshot`.

Only left one small note for an inconsistent environment variable name, which I consider a blocker.

File: scan/lib/scan/options.rb [hunk 2]
```diff
@@ -537,7 +537,12 @@ def self.available_options
                                     env_name: "SCAN_FAIL_BUILD",
                                     description: "Should this step stop the build if the tests fail? Set this to false if you're using trainer",
                                     type: Boolean,
-                                    default_value: true)
+                                    default_value: true),
+        FastlaneCore::ConfigItem.new(key: :package_authorization_provider,
```
Comment (lucgrabowski) [inline scan/lib/scan/options.rb:541]: Could you add a `verify_block` to config items (in scan, snapshot and gym) to check that the value is either `keychain` or `netrc` in a similar way as it's done for `catalyst_platform` above?

**Review by lucgrabowski — CHANGES_REQUESTED** [review]
Many thanks 🙏 for adding this fix 🚀 , I've added one comment. Could you please take a look at it :)?

**Review by revolter — APPROVED** [review]

File: scan/lib/scan/options.rb (hunk 2 above)
Comment (yarneo) [reply scan/lib/scan/options.rb:541]: thanks for the feedback! Done and tested to verify it works

**Comment by yarneo** [comment]
Appreciate the reviews 🙏

**Review by lucgrabowski — APPROVED** [review]

**Pull request merged**